File: go/src/344.go
func reverseString(s []byte) {
	n := len(s)
	for i := 0; i < n/2; i++ {
		swap(s, i, n-1-i)
	}
}

func swap(s []byte, i, j int) {
	tmp := s[i]
	s[i] = s[j]
	s[j] = tmp
}

File: py3/344.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        # two pointers
        i, j = 0, len(s)-1
        while i < j:
            # swap s[i] and s[j]
            tmp = s[i]
            s[i] = s[j]
            s[j] = tmp
            i += 1
            j -= 1
        